light/light_test.go

```diff
@@ -167,3 +167,76 @@ func waitForBlock(ctx context.Context, p provider.Provider, height int64) (*type
 		}
 	}
 }
+func TestClientStatusRPC(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	conf, err := rpctest.CreateConfig(t.Name())
+	require.NoError(t, err)
+
+	// Start a test application
+	app := kvstore.NewApplication()
+
+	_, closer, err := rpctest.StartTendermint(ctx, conf, app, rpctest.SuppressStdout)
+	require.NoError(t, err)
+	defer func() { require.NoError(t, closer(ctx)) }()
+
+	dbDir, err := os.MkdirTemp("", "light-client-test-status-example")
+	require.NoError(t, err)
+	defer os.RemoveAll(dbDir)
+
+	chainID := conf.ChainID()
+
+	primary, err := httpp.New(chainID, conf.RPC.ListenAddress)
+	require.NoError(t, err)
+
+	witnesses := []provider.Provider{primary}
+	// give Tendermint time to generate some blocks
+	block, err := waitForBlock(ctx, primary, 2)
+	require.NoError(t, err)
+
+	db, err := dbm.NewGoLevelDB("light-client-db", dbDir)
+	require.NoError(t, err)
+
+	c, err := light.NewClient(ctx,
+		chainID,
+		light.TrustOptions{
+			Period: 504 * time.Hour, // 21 days
+			Height: 2,
+			Hash:   block.Hash(),
+		},
+		primary,
+		witnesses,
+		dbs.New(db),
+		light.Logger(log.TestingLogger()),
+	)
+	require.NoError(t, err)
+
+	defer func() { require.NoError(t, c.Cleanup()) }()
+
+	lightStatus := c.Status(ctx)
+	require.NoError(t, err)
+	// Verify primary IP
+	require.True(t, lightStatus.Primary == primary.String())
+
+	// Verify IPs of witnesses
+	// ToDo - Add test with multiple witnesses
```

Comment:
```suggestion
	// TODO(jmalicevic): Add test with multiple witnesses
```

or maybe an issue number if it's a bigger project.

Reply:
I added an artificial test where the witness list has the primary multiple times. However, this should be a bug, we should not be able to create a light client whose witness list has essentially only the primary.